src/ui/main.cpp

```diff
@@ -35,14 +35,24 @@ struct Arguments {
     static optional<Arguments> parse(const QCommandLineParser& parser) {
         Arguments instance;
         auto args = parser.positionalArguments();
+        instance.export_ = parser.isSet("export");
+
         if (args.isEmpty()) {
+            if (instance.export_) {
+                qCritical() << QApplication::translate("main", "No file given to export.");
+                exit(1);
+            }
             return {};
         }
 
+        if (args.size() > 1) {
+            qWarning() << QApplication::translate(
+                "main", "Too many positional arguments given. Only the first one is used.");
```

Comment:
I would rather stop here: the user most likely made a mistake and the app is not going to do what they expect.